src/model.cpp

```diff
@@ -475,6 +475,45 @@ bool Model::hasImports() const
     return importsPresent;
 }
 
+void fixComponentUnits(const ModelPtr &model, const ComponentPtr &component)
+{
+    for (size_t v = 0; v < component->variableCount(); ++v) {
+        auto variable = component->variable(v);
+        if (variable->units() != nullptr) {
+            // Find the units in the model and switch out.
+            auto units = model->units(variable->units()->name());
```

Comment:
Not keen on using `name` to define the "same" units. Need to actually compare the units definitions...pretty sure there is existing functionality that does that.

Reply:
I wondered about that, except that according to the spec you can't have duplicate names, but you can have duplicate definitions.  Since the units are added to variables in the CellML format by their names, I figured this was the safer way to do it?

Reply:
Some may decide to clone something in a not-yet-valid model, no?

Reply:
Sure.  But that doesn't mean identifying units by their contents is correct either.  Since we actually use the names to identify things, I think it's better to continue with that philosophy. Duplicates or missing names should raise a warning though.

Reply:
in a valid model you can't have multiple units with the same name...

Reply:
Sure, but here I am talking about a model that 1) is not yet valid, 2) has two units with the same name, and 3) then you do the cloning.

Reply:
I agree, but I can't see how that issue can be easily addressed. While this PR will fix the issue of cloning resulting in multiple copies of the same units (making a valid model invalid).